File: packages/dart/lib/src/ubigeo/is_valid_ubigeo.dart
import 'ubigeo_store.dart';

bool isValidUbigeo(String code) {
  final digitsOnly = RegExp(r'^\d+$');
  if (!digitsOnly.hasMatch(code)) return false;
  switch (code.length) {
    case 2:
      return UbigeoStore.departments.containsKey(code);
    case 4:
      return UbigeoStore.provinces.containsKey(code);
    case 6:
      return UbigeoStore.districts.containsKey(code);
    default:
      return false;
  }
}

File: packages/python/peru_spatial_utils/ubigeo/is_valid_ubigeo.py
from ._records import departments, districts, provinces


def is_valid_ubigeo(code: str) -> bool:
    if not code.isdigit():
        return False
    length = len(code)
    if length == 2:
        return code in departments()
    if length == 4:
        return code in provinces()
    if length == 6:
        return code in districts()
    return False
